R/generateFilterValues.R

```diff
@@ -161,61 +161,72 @@ print.FilterValues = function(x, ...) {
 #' @param fvalues ([FilterValues])\cr
 #'   Filter values.
 #' @param sort (`character(1)`)\cr
-#'   Sort features like this.
-#'   \dQuote{dec} = decreasing, \dQuote{inc} = increasing, \dQuote{none} = no sorting.
+#'   Available options are:
+#'   - `"dec"`-> descending
+#'   - `"inc"` -> increasing
+#'   - `"none"` -> no sorting
+#'
 #'   Default is decreasing.
 #' @param n.show (`integer(1)`)\cr
 #'   Number of features (maximal) to show.
-#'   Default is 20.
+#'   Default is to plot all features.
 #' @param feat.type.cols (`logical(1)`)\cr
-#'   Colors for factor and numeric features.
-#'   `FALSE` means no colors.
-#'   Default is `FALSE`.
+#'   Whether to color different feature types (e.g. numeric | factor).
+#'   Default is to use no colors (`feat.type.cols = FALSE`).
 #' @template arg_facet_nrow_ncol
 #' @template ret_gg2
 #' @export
 #' @examples
 #' fv = generateFilterValuesData(iris.task, method = "variance")
 #' plotFilterValues(fv)
-plotFilterValues = function(fvalues, sort = "dec", n.show = 20L, feat.type.cols = FALSE, facet.wrap.nrow = NULL, facet.wrap.ncol = NULL) {
+plotFilterValues = function(fvalues, sort = "dec", n.show = nrow(fv$data),
```

Comment:
Don't think that this will work -- I would set a dummy value (`NULL`) as default and then check that inside the function to get the number of features in the data.

Reply:
It works, I just had a typo in the arg. There is no problem with setting a default arg dependent on another arg of a function.

I don't think setting the default to `NULL` has any advantage here: We are not doing any if-branching on this arg down in the function and just use it as an integer value for subsetting.
In general setting args to `NULL` by default is the preferred way in many cases but there are cases like this where deviating from this default makes sense (imo).

Reply:
Ok thanks.